### PR TITLE
Add tx handler

### PR DESCRIPTION
**Pull request by marktrs**
extract migration tx handler from #1224
go-loom : loomnetwork/go-loom#446

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ BINANCE_TGORACLE_DIR=$(GOPATH)/src/$(PKG_BINANCE_TGORACLE)
 # NOTE: To build on Jenkins using a custom go-loom branch update the `deps` target below to checkout
 #       that branch, you only need to update GO_LOOM_GIT_REV if you wish to lock the build to a
 #       specific commit.
-GO_LOOM_GIT_REV = HEAD
+GO_LOOM_GIT_REV = add-tx-handler
```
Comment (enlight) [inline Makefile]: I've merged the go-loom PR, can set this back to HEAD

File: tx_handler/migration_tx_handler_test.go [hunk 2]
```diff
@@ -29,9 +30,9 @@ func TestMigrationTxHandler(t *testing.T) {
 	migrationTx1 := mockMessageTx(t, uint32(1), origin, origin)
 
 	migrationFuncs := map[int32]MigrationFunc{
-		1: func(ctx *migrations.MigrationContext) error { return nil },
-		2: func(ctx *migrations.MigrationContext) error { return nil },
-		3: func(ctx *migrations.MigrationContext) error { return nil },
+		1: func(ctx *migrations.MigrationContext, parameters []byte) error { return nil },
+		2: func(ctx *migrations.MigrationContext, parameters []byte) error { return nil },
+		3: func(ctx *migrations.MigrationContext, parameters []byte) error { return nil },
```
Comment (enlight) [inline tx_handler/migration_tx_handler_test.go]: Update one of these functions to expect the parameters to match a particular value, and update the tests below to pass that value.

File: tx_handler/migration_tx_handler.go [hunk 3]
```diff
@@ -86,13 +87,12 @@ func (h *MigrationTxHandler) ProcessTx(
...
 	}
-
 	migrationCtx := migrations.NewMigrationContext(h.Manager, h.CreateRegistry, state, origin)
-	if err := migrationFn(migrationCtx); err != nil {
+	if err := migrationFn(migrationCtx, tx.Input); err != nil {
 		return r, errors.Wrapf(err, "migration %d failed", int32(tx.ID))
 	}
 
-	state.Set(migrationKey(tx.ID), msg.Data)
+	state.Set(migrationKey(tx.ID), []byte{1})
```
Comment (pathornteng) [inline tx_handler/migration_tx_handler.go]: Why is this changed? If you want to change this to save some space, a feature flag is needed or @enlight approve this.

Comment (marktrs) [reply tx_handler/migration_tx_handler.go]: yep @enlight approved 
https://github.com/loomnetwork/loomchain/pull/1431#pullrequestreview-278234687

Comment (pathornteng) [reply tx_handler/migration_tx_handler.go]: Approved with a feature flag lol, so you need to wrap this with a feature flag to prevent consensus failure.

Comment (enlight) [reply tx_handler/migration_tx_handler.go]: Since migration txs can have input parameters now it doesn't make much sense to store all that in the app store, we only use this key to check if a migration has already ran, the value we store here is irrelevant.

This change does need a new feature flag.
